test/integration/loans.test.ts

```diff
@@ -165,6 +172,167 @@ describe('Loans', () => {
             })
+
+            it('should be able to create and repay a loan', async () => {
+              const borrower = await getNamedSigner('borrower')
+
+              const { getHelpers } = await takeOutLoanWithNfts(hre, {
+                amount: 100, //should use raw amt not formatted.. oh well
+                lendToken: market.lendingToken,
+                borrower: borrower,
+                version: 1,
+              })
+
+              const helpers: LoanHelpersReturn = await getHelpers()
+
+              expect(helpers.details.loan).to.exist
+
+              console.log('helpers.details.loan', helpers.details.loan)
+
+              const loanId = helpers.details.loan.id
+              console.log('loanId', loanId)
+
+              //  const lHelpers = await loanHelpers(loanId)
+
+              //need to give fake ERC20 to borrower
+              //and add fake ERC20 to the borrow escrow
+
+              const lendingToken =
+                typeof market.lendingToken === 'string'
+                  ? await tokens.get(market.lendingToken)
+                  : market.lendingToken
+
+              const lendingTokenDecimals = await lendingToken.decimals()
+              console.log('decimals', lendingTokenDecimals)
+
+              await getFunds({
+                to: borrower,
+                tokenSym: market.lendingToken,
+                amount: (200 * 10 ** lendingTokenDecimals).toString(),
+                hre,
+              })
+
+              const borrowerAddress = await borrower.getAddress()
+              const borrowerBalance = await lendingToken.balanceOf(
+                borrowerAddress
+              )
+
+              console.log('balanceOf', borrowerBalance.toString())
+
+              console.log('market.lendingToken', market.lendingToken)
+
+              const balanceLeftToRepay = await diamond.getTotalOwed(loanId)
+              console.log('balanceLeftToRepay', balanceLeftToRepay)
+
+              await lendingToken
+                .connect(ethers.provider.getSigner(borrowerAddress))
+                .approve(diamond.address, balanceLeftToRepay)
+
+              //let response = await helpers.repay( 100000000 , borrower )
+
+              await diamond
+                .connect(borrower)
+                .repayLoan(
+                  loanId,
+                  (100 * 10 ** lendingTokenDecimals).toString()
+                )
+
+              const totalOwedAfterRepay = await diamond.getTotalOwed(loanId)
+
+              console.log('totalOwedAfterRepay', totalOwedAfterRepay.toString())
+
+              const loanData = await diamond.getLoan(loanId)
+              console.log('loanData')
+
+              console.log('balanceLeftToRepay 2', balanceLeftToRepay)
+
+              console.log('balanceOf After', borrowerBalance.toString())
+
+              expect(parseInt(borrowerBalance.toString())).to.equal(
+                200 * 10 ** lendingTokenDecimals
+              )
```

Comment:
you aren't updating this value after the repay so the console log comment is a bit confusing. also since you used `getFunds` for this same amount and the value is not being updated, this check seems to just be verifying that you indeed received this amount from the `getFunds` call. 

should also use `borrowerBalance.should.eql(...)`